modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2939,6 +2995,12 @@ else if (StringUtil.equals(
 				throw new ObjectEntryValuesException.ListTypeEntry(
 					entry.getKey());
 			}
+
+			if (objectEntry != null) {
```

Comment:
@RenanUmV

before to validate we need to check if the objectField is state, because this validation should be only applied for picklist fields that are states.

So incremente this conditional below:

`&& objectField.isState()`

Thank you!